src/core/model_config_utils.h

```diff
@@ -104,33 +102,33 @@ Status BuildEnsembleGraph(
 /// Validate that input is specified correctly in a model
 /// configuration.
 /// \param io The model input.
+/// \param max_batch_size The max batch size specified in model configuration.
 /// \return The error status. A non-OK status indicates the input
 /// is not valid.
-Status ValidateModelInput(const ModelInput& io);
+Status ValidateModelInput(const ModelInput& io, int32_t max_batch_size);
 
-/// Validate that an input is specified correctly in a model
-/// configuration and matches one of the allowed input names.
+/// Validate that an input matches one of the allowed input names.
 /// \param io The model input.
 /// \param allowed The set of allowed input names.
 /// \return The error status. A non-OK status indicates the input
 /// is not valid.
-Status ValidateModelInput(
+Status AllowModelInput(
```

Comment:
Can you change name to "CheckAllowedModelInput" so it is clear that this is a validation and not an assignment.